### ISSUE TITLE
events.closeAll does not really seem closing all the channels

### ISSUE THREAD

**Issue by tmokmss** [aws-amplify/amplify-js#13984]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

GraphQL API

### Amplify Version

v6

### Amplify Categories

api

### Backend

Other

### Environment information

<details>

```
# Put output below this line
```

</details>

### Describe the bug

`events.closeAll` does not seem workiing as expected. Even after calling the function, all the existing subscription remain active and new events are subscribed.

### Expected behavior

After calling `event.closeAll`, all the existing channels are closed and no new events are subscribed.

### Reproduction steps

Bascally I followed this tutorial: https://docs.aws.amazon.com/appsync/latest/eventapi/build-amplify-app.html#deploy-react-vite-app

But when I changed a line as below, channels will not be closed at all.

```diff
    return () => {
-      pr?.then((channel) => channel?.close())
+      events.closeAll()
    }
```

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### Manual configuration

```js
  {
    API: {
      Events: {
        endpoint: "https://redacted.appsync-api.us-west-2.amazonaws.com/event",
        defaultAuthMode: "apiKey",
        apiKey: "da2-redacted",
      },
    },
  };
```

**Comment by chrisbonifacio** [comment]
thank you for raising this issue! i've marked as a bug for the team.

we will look into it and report back with any progress on a fix.

**Comment by iartemiev** [comment]
@tmokmss - we published a bug fix for this issue to `aws-amplify@unstable`. It will be included in the next library release next week.

**Comment by iartemiev** [comment]
The bug fix has been released with `aws-amplify@6.8.1`.
